src/thok.rs

```diff
@@ -234,4 +241,34 @@ impl Thok {
 
         Ok(())
     }
+
+    pub fn get_skip_count(&mut self, max_width: usize) {
```

Comment:
Would you mind adding some comments in this fn?

Reply:
Added. Its been a while since I've looked at the code myself, but hopefully the comments explain the algorithm well enough.